File: natour-backend/src/main/java/com/natour/natour/model/entity/ChatMessage.java
package com.natour.natour.model.entity;


import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.springframework.lang.Nullable;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "NT_CHAT_MESSAGES")
public class ChatMessage {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(cascade=CascadeType.ALL)
    private Chat chat;

    @Nullable
    private Long idTrail;

    private String usernameOwner;
    private String date;
    private String message;
    private String time;
}

File: natour-android/app/src/main/java/com/example/natour/data/model/ChatMessage.kt
package com.example.natour.data.model

import kotlinx.serialization.Serializable

@Serializable
data class ChatMessage(
    val idChat: Long,
    val idTrail: Long,
    val usernameOwner: String,
    val usernameRecipient: String?,
    val date: String,
    val time: String,
    val message: String
)
